### PR TITLE
Use taxonomy to fetch small units on search

### PR DESCRIPTION
**Pull request by vforge**
## Links

* http://wikia-inc.atlassian.net/browse/CAKE-5274

## Description

- Use taxonomy to drive small unit on search

## Reviewers

@Wikia/cake

### CODE REVIEW

File: app/services/affiliate-slots.js [hunk 1]
```diff
@@ -77,6 +77,35 @@ const checkMobileSystem = (unit) => {
...
+ *
+ * @param {Object} response
+ * @returns {Targeting[]}
+ */
+const flattenKnowledgeGraphTargeting = (response) => {
+  // convert API to nicer, more useful format
+  const targeting = [];
+
+  // convert from tree structure to flat structure for easier comparison later
+  response.forEach((e) => {
```
Comment (mklucsarits) [inline app/services/affiliate-slots.js]: Can you use more descriptive variables than single letters?

File: app/services/affiliate-slots.js [hunk 2]
```diff
@@ -175,16 +204,41 @@ export default Service.extend({
...
+              .filter(u => !u.isBig)
+              // filter units disabled on article page
+              .filter(u => !u.disableOnSearch);
+
+            // fetch only the first unit if available
+            resolve(availableUnits.length > 0 ? availableUnits[0] : undefined);
+          })
+          .catch((error) => {
+            // log and do not raise anything
+            console.error(error);
```
Comment (mklucsarits) [inline app/services/affiliate-slots.js]: I think the linter might object to `console` statements.

Comment (vforge) [reply app/services/affiliate-slots.js]: d3f3720

File: app/services/affiliate-slots.js (hunk 1 above)
Comment (vforge) [reply app/services/affiliate-slots.js]: d3f3720

**Review by mklucsarits — APPROVED** [review]
Code looks good to me, and tested

**Pull request merged**